assets/js/modules/analytics-4/components/widgets/LeastEngagingPagesWidget.js

```diff
@@ -0,0 +1,240 @@
+	MetricTileTable,
+	MetricTileTablePlainText,
+} from '../../../../components/KeyMetrics';
+import Link from '../../../../components/Link';
+import { ZeroDataMessage } from '../../../analytics/components/common';
+import { numFmt } from '../../../../util';
+import whenActive from '../../../../util/when-active';
+import ConnectGA4CTATileWidget from './ConnectGA4CTATileWidget';
+import useViewOnly from '../../../../hooks/useViewOnly';
+const { useSelect, useInViewSelect } = Data;
+
+function LeastEngagingPagesWidget( props ) {
+	const { Widget } = props;
+
+	const viewOnlyDashboard = useViewOnly();
+
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} )
+	);
+
+	const pageViewsReportOptions = {
+		...dates,
+		dimensions: [ 'pagePath' ],
+		metrics: [ { name: 'screenPageViews' } ],
+		orderby: [
+			{
+				metric: { metricName: 'screenPageViews' },
+				desc: true,
+			},
+		],
+	};
+
+	const pageViewsReport = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( pageViewsReportOptions )
+	);
+
+	const medianIndex = parseInt( pageViewsReport?.rowCount / 2, 10 );
+	const medianPageViews =
+		parseInt(
+			pageViewsReport?.rows?.[ medianIndex ]?.metricValues?.[ 0 ]?.value,
+			10
+		) || 0;
+
+	const reportOptions = {
+		...dates,
+		dimensions: [ 'pagePath' ],
+		metrics: [ 'bounceRate', 'screenPageViews' ],
+		orderby: [
+			{
+				metric: { metricName: 'bounceRate' },
+				desc: true,
+			},
+			{
+				metric: { metricName: 'screenPageViews' },
+				desc: true,
+			},
+		],
+		metricFilters: {
+			screenPageViews: {
+				operation: 'GREATER_THAN_OR_EQUAL',
+				value: { int64Value: medianPageViews },
+			},
+		},
+		limit: 3,
+	};
+
+	const loadedPageViewsReport = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).hasFinishedResolution( 'getReport', [
+			pageViewsReportOptions,
+		] )
+	);
+
+	const pageViewsReportError = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
+			pageViewsReportOptions,
+		] )
+	);
+
+	const report = useSelect( ( select ) => {
+		if ( loadedPageViewsReport && ! pageViewsReportError ) {
+			return select( MODULES_ANALYTICS_4 ).getReport( reportOptions );
+		}
+	} );
+
+	const error = useSelect( ( select ) => {
+		const reportError = select( MODULES_ANALYTICS_4 ).getErrorForSelector(
+			'getReport',
+			[ reportOptions ]
+		);
+
+		if ( pageViewsReportError && reportError ) {
+			return [ pageViewsReportError, reportError ];
+		}
+
+		return pageViewsReportError || reportError || undefined;
+	} );
+
+	const titles = useInViewSelect( ( select ) =>
+		! error
+			? select( MODULES_ANALYTICS_4 ).getPageTitles(
+					report,
+					reportOptions
+			  )
+			: undefined
+	);
+
+	const loading = useSelect(
+		( select ) =>
+			! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+				'getReport',
+				[ reportOptions ]
+			) ||
+			titles === undefined ||
+			report === null ||
```

Comment:
Prior to the amendment above, AFAIK we wouldn't have been expecting `report` to be `null` here (sorry I didn't catch this the first time around!).

Although I've suggested we _do_ now return `null` above, it's for the error case rather than loading, so I think we can still remove this line.

```suggestion
```